### PR TITLE
Enable langversion parsing for script command line

### LINKED ISSUE #41893: How to specify langversion from csi
Is there any way to enable it or it's deliberately unsupported?

Ideally we should only write `csi /langversion:preview` to enable preview features.

### PR DESCRIPTION
**Pull request by alrz**
Closes #41893

### CODE REVIEW

**Comment by alrz** [comment]
@tmat Is there any other option that we may want to enable for csi? unsafe maybe?

**Comment by alrz** [comment]
❓ Is there any way to make csi print IL code instead of program output? That would be super useful for trying out new language features and manual testing.

**Comment by tmat** [comment]
> Is there any way to make csi print IL code instead of program output? That would be super useful for trying out new language features and manual testing.

No, that's not a feature that would be useful for most of our customers. You can use [sharplab.io](http://sharplab.io) to do that.

**Comment by alrz** [comment]
@tmat Can I get a review on this please? thanks.

File: src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs [hunk 1]
```diff
@@ -78,6 +78,12 @@ private int RunInteractiveCore(ErrorLogger errorLogger)
                 return 0;
             }
 
+            if (_compiler.Arguments.DisplayLangVersions)
+            {
+                _compiler.PrintLangVersions(_console.Out);
```
Comment (jcouv) [inline src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs:83]: Do we have a test that harnesses this change?

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 2]
```diff
@@ -1423,10 +1423,7 @@ public void ArgumentParsing()
             Assert.False(parsedArgs.SourceFiles.Any());
 
             parsedArgs = CSharpCommandLineParser.Script.Parse(new[] { "/langversion:?" }, WorkingDirectory, sdkDirectory);
-            parsedArgs.Errors.Verify(
-                // error CS2007: Unrecognized option: '/langversion:?'
-                Diagnostic(ErrorCode.ERR_BadSwitch).WithArguments("/langversion:?").WithLocation(1, 1)
-                );
+            parsedArgs.Errors.Verify();
```
Comment (alrz) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:1426]: @jcouv I supposed this is sufficient. Do we need to assert anything else?

Comment (jcouv) [reply src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:1426]: The change I referred to *adds* something (we print more info), but this test shows that we removed an error. The test is not covering that particular piece of code.

File: src/Scripting/CSharpTest/CommandLineRunnerTests.cs [hunk 3]
```diff
@@ -830,6 +830,29 @@ public void HelpCommand()
 > ", runner.Console.Out.ToString());
         }
 
+        [Fact]
+        public void LangVersions()
```
Comment (alrz) [inline src/Scripting/CSharpTest/CommandLineRunnerTests.cs:834]: Same as `LangVersion_ListLangVersions` for csi

**Comment by alrz** [comment]
ping @tmat

**Comment by alrz** [comment]
@dotnet/roslyn-interactive

**Comment by alrz** [comment]
Apparently I can't tag teams @dotnet/roslyn-interactive

**Comment by tmat** [comment]
>  Is there any other option that we may want to enable for csi?

I don't think we do. csi shouldn't have too many command line switches. More switches the more difficult it is to write self-contained script files.

**Comment by tmat** [comment]
@jcouv Needs 2nd compiler team review.

**Comment by jcouv** [comment]
Tagging @dotnet/roslyn-compiler for second review of community PR. Thanks

**Comment by jcouv** [comment]
Merged/squashed. Thanks @alrz!

**Pull request merged**